systemd/systemd.go

```diff
@@ -563,6 +614,51 @@ func (s *systemd) StartNoBlock(serviceNames []string) error {
 	return err
 }
 
+// Check if unit is in failed state
+func (s *systemd) IsFailed(serviceNames []string) bool {
+	var err error
+	var out []byte
+	if s.rootDir != "" {
+		out, err = s.systemctl(append([]string{"--root", s.rootDir, "is-failed"}, serviceNames...)...)
+	} else {
+		out, err = s.systemctl(append([]string{"is-failed"}, serviceNames...)...)
+	}
```

Comment:
nitpick: I'd only do the parameters composition in the `if`, and invoke `systemctl` outside of it. Then you don't need to declare the variables above.

Reply:
There's a bunch of code just above which uses this pattern. I'm ok with tweaking it, but I'd rather change all of the code, so a followup then?

Reply:
Sure, and it's a nitpick anyway :-)